README.md

```diff
@@ -28,6 +32,121 @@ Project is written in TypeScript and provides type safety out of the box. No Flo
 -   [Server Side Rendering with `useStaticRendering`](#server-side-rendering-with-usestaticrendering)
 -   [Why no Provider/inject?](#why-no-providerinject)
 
+## Overview
+
+This library has two goals:
+
+1. Make it possible to introduce reactivity to React function components
+2. Make it possible to have local observable state in React function components
+
+This library does _not_ support class based components. If you need class based component support, use [mobx-react](https://github.com/mobxjs/mobx-react), which wraps this library.
+
+### Adding reactivity to React components
+
+There are three ways to add reactivity to components: `observer(fn)`, `<Observer>{renderFn}</Observer>` and `useObserver(renderFn)`.
+In most cases, there difference doesn't matter that much, but here is an overview of the three different approaches:
+
+```javascript
+import { observable } from "mobx"
+import { Observer, useObserver, observer }  from "mobx-react-lite"
+
+const person = observable({
+    name: "John"
+})
+
+const P1 = observer(function P1({ person }) {
+    return <h1>{person.name}</h1>
+})
+
+const P2 = ({ person }) => (
+    <Observer>{() => <h1>{person.name}</h1>}</Observer>
+)
+
+const P3 = ({ person }) => {
+    return useObserver(() => (
+        <h1>{person.name}</h1>
+    ))
+})
+
+ReactDOM.render(<div>
+    <P1 person={person} />
+    <P2 person={person} />
+    <P3 person={person} />
+</div>)
+
+setTimeout(() => {
+    person.name = "Jane"
+}, 1000)
+```
+
+When using `observer`, the component `props` will be made automatically observable (so that they can be used in reactions and computed values), and `React.memo` will be applied automatically to the comment.
+
+`useObserver` keeps the component tree flat as it doesn't introduce a higher order component, but to make props observable, the [`useAsObservableSource`](#useasobservablesourcetstate-t-t) hook should be used.
+
+`<Observer>` is slightly better optimized compared to `useObserver`, but does introduce an additional component in the component tree.
+
+### Introducing local observable state
+
+Local observable state can be introduced by using the [`useLocalStore`](#uselocalstoretinitializer---t-t) hook, that runs once to create an observable store. A quick example would be:
+
+```javascript
+import { useLocalStore, useObserver } from "mobx-react-lite"
+
+const P3 = () => {
+    const todo = useLocalStore(() => ({
+        title: "Test",
+        done: true,
+        toggle() {
+            this.done = !this.done
```

Comment:
I have a feeling we should figure out what's wrong with the TypeScript there. I don't feel like recommending `this` in here and then people coming complaining it's not working.

Also honestly, if we forget about TypeScript, isn't it actually better for a user to see `todo.done` in here? I mean these are functional components, one of the selling points is no need for confusing `this`. Why should possibly "scare" people off?

Reply:
afaik TS inference is normally better with `this`, and hopefully this hints that for examle `useLocalStore(() => new CounterStore())` works, as does extracting that whole function outside the component to avoid re-creating the closure over and over again.

Probably we should add that as tip as well.

Reply:
Ok, so what are we going to do about it? All examples should use a unified way, otherwise people come asking sometimes it's `this` or `that`. I vote for `that` for reasons I've said above.